### DISCUSSION TITLE
Any way to check if Metamask/Connected wallet is Deleted and update wallet connection state realtime

### DISCUSSION THREAD

**Discussion by ShahzaibAyyub** [WalletConnect/walletconnect-monorepo#2469] [Developer Support] [answered]
Is there any way to know the state of wallet connection when the wallet app is deleted. My client reported a bug in which when wallet is connected through web3modal, if the user deletes the wallet app suppose Metamask, then the state of wallet connection is shown as connected. Is there any way to update the wallet state when this happens. I'm using   "const { disconnect } = useDisconnect();" for wallet disconnection from wagmi, but it can only be used when invoked. Here are the package versions I'm using:

    "@web3modal/ethereum": "2.2.1",
    "@web3modal/react": "2.2.1",
    "wagmi": "^0.12.12",

**Comment by xzilja** [comment, marked as answer]
I'm afraid not, unless wallet itself can disconnect the session before being deleted. You can ping the wallet, but if it is not active on users device, it will not reply.
